Lib/socket.py

```diff
@@ -728,6 +728,149 @@ def create_connection(address, timeout=_GLOBAL_DEFAULT_TIMEOUT,
+    AF_INET6 in case family can't be determined from *host*.
+
+    *type* should be either SOCK_STREAM or SOCK_DGRAM.
+
+    *backlog* is the queue size passed to socket.listen() and is ignored
+    for SOCK_DGRAM socket types.
+
+    *reuse_addr* and *reuse_port* dictate whether to use SO_REUSEADDR
+    and SO_REUSEPORT socket options.
+
+    *flags* is a bitmask for getaddrinfo().
+
+    *hybrid_ipv46*: if True and the platform supports it it will create
+    a socket able to accept both IPv4 and IPv6 connections.
+    When False it will explicitly disable this option on platforms that
+    enable it by default (e.g. Linux).
+
+    >>> with bind_socket((None, 8000)) as server:
+    ...     while True:
+    ...         conn, addr = server.accept()
+    ...         # handle new connection
+    """
+    host, port = address
+    if host == "":
+        # https://mail.python.org/pipermail/python-ideas/2013-March/019937.html
+        host = None  # all interfaces
+    # Note about Windows: by default SO_REUSEADDR is not set because:
+    # 1) It's unnecessary: bind() will succeed even in case of a
+    # previous closed socket on the same address and still in TIME_WAIT
+    # state.
+    # 2) If set, another socket will be free to bind() on the same
+    # address, effectively preventing this one from accepting connections.
+    # Also, it sets the process in a state where it'll no longer respond
+    # to any signals or graceful kills. The option can still be set though
+    # for SOCK_DGRAM multicast sockets.
+    # See: msdn2.microsoft.com/en-us/library/ms740621(VS.85).aspx
+    iswin = os.name in ('nt', 'cygwin')
+    if reuse_addr is None:
+        reuse_addr = not iswin and hasattr(_socket, 'SO_REUSEADDR')
+    elif reuse_addr and not hasattr(_socket, 'SO_REUSEADDR'):
+        raise ValueError("SO_REUSEADDR not supported on this platform")
+    elif reuse_addr and iswin and type != SOCK_DGRAM:
+        raise ValueError("SO_REUSEADDR allowed for SOCK_DGRAM type only")
+
+    if reuse_port and not hasattr(_socket, "SO_REUSEPORT"):
+        raise ValueError("SO_REUSEPORT not supported on this platform")
+    if type not in (SOCK_STREAM, SOCK_DGRAM):
+        raise ValueError("only SOCK_STREAM and SOCK_DGRAM types are supported")
+    if flags is None:
+        flags = AI_PASSIVE
+    if hybrid_ipv46:
+        if type != SOCK_STREAM:
+            raise ValueError("hybrid_ipv46 requires SOCK_STREAM type")
+        if family not in {AF_INET6, AF_UNSPEC}:
+            raise ValueError("hybrid_ipv46 requires AF_INET6 family")
+        if not supports_hybrid_ipv46():
+            raise ValueError("hybrid_ipv46 not supported on this platform")
+        family = AF_INET6
+
+    info = getaddrinfo(host, port, family, type, 0, flags)
+    if family == AF_UNSPEC:
+        # prefer AF_INET over AF_INET6
+        info.sort(key=lambda x: x[0] == AF_INET, reverse=True)
+
+    err = None
+    for res in info:
+        af, socktype, proto, canonname, sa = res
```

Comment:
may rename "sa" to "sock_addr" or "address".

Reply:
This is the same name used in `create_connection` and also in socket module's doc example. Better we stay consistent.